scripts/protein/esm2/esm2_infer.py

```diff
@@ -0,0 +1,274 @@
+        include_logits=args.include_logits,
+        micro_batch_size=args.micro_batch_size,
+        accumulate_grad_batches=args.accumulate_grad_batches,
+        precision=args.precision,
+        tensor_model_parallel_size=args.tensor_model_parallel_size,
+        pipeline_model_parallel_size=args.pipeline_model_parallel_size,
+        devices=args.num_gpus,
+        num_nodes=args.num_nodes,
+        config_class=args.config_class,
+    )
+
+
+def get_parser():
+    """Return the cli parser for this tool."""
+    parser = argparse.ArgumentParser(description="Infer ESM2.")
+    parser.add_argument(
+        "--checkpoint-path",
+        type=Path,
+        required=True,
+        help="Path to the ESM2 pretrained checkpoint",
+    )
+    parser.add_argument(
+        "--data-path",
+        type=Path,
+        required=True,
+        help="Path to the CSV file containing sequences and label columns",
+    )
+    parser.add_argument("--results-path", type=Path, required=True, help="Path to the results file.")
+
+    parser.add_argument(
+        "--precision",
+        type=str,
+        choices=get_args(PrecisionTypes),
+        required=False,
+        default="bf16-mixed",
+        help="Precision type to use for training.",
+    )
+    parser.add_argument(
+        "--num-gpus",
+        type=int,
+        required=False,
+        default=1,
+        help="Number of GPUs to use for training. Default is 1.",
+    )
+    parser.add_argument(
+        "--num-nodes",
+        type=int,
+        required=False,
+        default=1,
+        help="Number of nodes to use for training. Default is 1.",
+    )
+    parser.add_argument(
+        "--micro-batch-size",
+        type=int,
+        required=False,
+        default=2,
+        help="Micro-batch size. Global batch size is inferred from this.",
+    )
+    parser.add_argument(
+        "--pipeline-model-parallel-size",
+        type=int,
+        required=False,
+        default=1,
+        help="Pipeline model parallel size. Default is 1.",
+    )
+    parser.add_argument(
+        "--tensor-model-parallel-size",
+        type=int,
+        required=False,
+        default=1,
+        help="Tensor model parallel size. Default is 1.",
+    )
+    parser.add_argument(
+        "--include-hiddens",
+        action="store_true",
+        default=False,
+        help="Include hiddens in output of inference",
+    )
+    parser.add_argument(
+        "--include-embeddings",
+        action="store_true",
+        default=False,
+        help="Include embeddings in output of inference",
+    )
+    parser.add_argument(
+        "--include-logits", action="store_true", default=False, help="Include per-token logits in output."
+    )
+    parser.add_argument(
+        "--accumulate-grad-batches",
+        type=int,
+        required=False,
+        default=1,
+        help="Gradient accumulation steps. Global batch size is inferred from this.",
+    )
```

Comment:
What does gradient accumulation do for inference? Can we remove this?

Reply:
hm, yeah good catch, will remove that